File: ACT2.py
class Character:
    def __init__(self, name, class_type, level):
        self.__name = name
        self.__class_type = class_type
        self.__level = level
        
    def get_name(self):
        return self.__name
        
    def get_class_type(self):
        return self.__class_type
        
    def get_level(self):
        return self.__level
        
    def level_up(self):
        self.__level += 1
        return self.__level
        print(f"{self.__name} has leveled up to level {self.__level}")
        
    def display_info(self):
        print(f"Name: {self.__name}")
        print(f"Class: {self.__class_type}")
        print(f"Level: {self.__level}")
        
if __name__ == "__main__":
    valid_classes = {"warrior", "mage", "archer"}
    
    name = input("Enter character name: ")
    class_type = input("Enter class type (warrior/mage/archer)")
    while True:
        if class_type not in valid_classes:
            break
        else:
            break
        
    level = int(input("Enter initial level:"))
    
    player = Character(name, class_type, level)
    
    print("\nCharacter details:")
    player.display_info()
    
    while True:
        choice = input("Do you want to level up? (yes/no)")
        if choice.lower() == "yes":
            player.level_up()
            print("Updated Character details:")
            player.display_info()
        elif choice.lower() == "no":
            player.display_info()
            break
        else:
            print("Invalid input!")
            
        
    
    
    

File: ACT2.java
import java.util.Scanner;

class Character{
    private String name;
    private String classType;
    private int level;
    
    public Character(String name, String classType, int level){
        this.name = name;
        this.classType = classType;
        this.level = level;
    }
    
    public String getName(){
        return name;
    }
    
    public String getClassType(){
        return classType;
    }
    
    public int getLevel(){
        return level;
    }
    
    public void levelUp(){
        level++;
        System.out.println(name + "has level up to level " + level);
    }
    
    public void displayInfo(){
        System.out.println("Name: " + name);
        System.out.println("Class: " + classType);
        System.out.println("Level: " + level);
    }
}

class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Enter character name: ");
        String name = scanner.nextLine();
        
        String classType;
        while (true){
            System.out.print("Enter class type (warrior/mage/archer): ");
            classType = scanner.nextLine().toLowerCase();
            
            if (!classType.equals("warrior") && !classType.equals("mage") && !classType.equals("archer")){
                System.out.println("Invalid Input");
            } else {
                break;
            }
            
        }
        
        System.out.print("Enter initial level: ");
        int level = scanner.nextInt();
        scanner.nextLine();
        
        Character player = new Character(name, classType, level);
        
        System.out.print("\nCharater details:");
        player.displayInfo();
        
        while (true){
            System.out.print("do you want to level up? (yes/no): ");
            String choice = scanner.nextLine().toLowerCase();
            if (choice.equals("yes")){
                player.levelUp();
                player.displayInfo();
            } else if (choice.equals("no")){
                player.displayInfo();
                break;
            } else {
                System.out.println("Invalid input");
            }
        }
        
    }
}
